internal/requestclient/http.go

```
@@ -106,10 +110,16 @@ func httpMiddleware(next http.Handler, external bool) http.Handler {
 			if cfIPCountry := req.Header.Get("CF-IPCountry"); cfIPCountry != "" && cfIPCountry != "XX" {
 				wafIPCountryCode = cfIPCountry
 			}
+
+			// Set the real IP of the client to the connecting IP, if using Cloudflare.
+			if forwardedFor != "" {
+				realIp = forwardedFor
+			}
 		}
 
 		ctxWithClient := WithClient(req.Context(), &Client{
-			IP:           strings.Split(req.RemoteAddr, ":")[0],
+			IP:           remoteAddr,
+			RealIP:       realIp,
```

Assuming there's a strong case for a new field `RealIP`, we should make this always empty outside of the CF case and omitted by default IMO, to avoid providing a misleading value